### PR TITLE
Compression Cleanup/Memory Fixes

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
- Updated klauspost/compress to 1.11.7
- Fixed zstd usage to prevent resource leaks
  - Added ResetReader/Writer to allow reuse of a compressed reader/writer
  - Updated PageReader/BufferedAppender to use reset methods
  - Added Close() to Iterator/PageReader to allow for cleanup
  - Made use of Close methods in all appropriate places
  - Updated default to zstd
  - Added docs re: compression

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/tempo/website/configuration/_index.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ This section explains the configuration options for Tempo as well as the details
   - [Compactor](#compactor)
   - [Storage](#storage)
   - [Memberlist](#memberlist)
+  - [Compression](#compression)
 
 ## Authentication/Server
 Tempo uses the Weaveworks/common server.  See [here](https://github.com/weaveworks/common/blob/master/server/server.go#L45) for all configuration options.
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/_index.md:21]:
```suggestion
Tempo uses the Weaveworks/common server.  Refer to [this page](https://github.com/weaveworks/common/blob/master/server/server.go#L45) for all configuration options.
```

**Review by achatterjee-grafana — APPROVED** [review]
Doc content looks good. Asked for one minor change.

File: docs/tempo/website/configuration/_index.md (hunk 1 above)
Comment (joe-elliott) [reply docs/tempo/website/configuration/_index.md:21]: This is not really in the scope of the PR i'm making.  I'd rather make this PR separately if we desire this change.

Comment (achatterjee-grafana) [reply docs/tempo/website/configuration/_index.md:21]: Sure, it is a documentation style issue about consistency, I can take care of it later.

**Review by mdisibio — APPROVED** [review]

**Pull request merged**